sycl/source/detail/queue_impl.cpp

```diff
@@ -99,10 +99,15 @@ event queue_impl::memset(const std::shared_ptr<detail::queue_impl> &Self,
   // Emit a begin/end scope for this call
   PrepareNotify.scopedNotify((uint16_t)xpti::trace_point_type_t::task_begin);
 #endif
+  // If we have a command graph set we need to capture the memset through normal
```

Comment:
It seems like this logic should be common for all shortcuts, i.e. any operation submitted while the queue is recording should go through the scheduler. Looks like we already have these checks for `mem_advise`, `memcpy` and, with this patch, `memset`. 
This leaves `memcpyToDeviceGlobal` and `memcpyFromDeviceGlobal`. This check should extend to them as well, right? If so, it should be made in `submitMemOpHelper`.

Reply:
Is this really necessary? There is not much duplicated code for the graph recording case and I don't really see how `submitMemOpHelper` could be extended cleanly to support this.

At a minimum we would have to pass a second lambda function to the call to `submitMemOpHelper` and I think this just hurts readability while providing only a minimal reduction in overall code:

```c++
  return submitMemOpHelper(
      Self, DepEvents,
      [&][handler &CGH) { CGH.memset(Ptr, Value, Count); },
      [](const auto &...Args) { MemoryManager::fill_usm(Args...); }, Ptr, Self,
      Count, Value);
```

Reply:
May be not for the graph case specifically, but `submitMemOpHelper` already has logic for submitting via the handler in other cases and we're already passing a second lambda there, see https://github.com/intel/llvm/blob/eead9892edfe2bf9e7a8f501ff86e545d1f786e4/sycl/source/detail/queue_impl.cpp#L127-L130

Reply:
Sorry, we missed this (we were working with a version of `queue_impl.cpp` prior to commit https://github.com/intel/llvm/commit/f3d30b6f2e43734c825ca4c5dcd0c3bb6570ed67 ). The code has been updated and the specific case for graph is handled in `submitMemOpHelper` now.